kansha/board/comp.py

```diff
@@ -886,6 +898,15 @@ def reload_search(self):
         self.must_reload_search = False
         return self.search(self.last_search)
 
+    def save_as_template(self, shared):
+        data = self.data.to_template(shared)
+        fname = unicodedata.normalize('NFKD', self.data.title).encode('ascii', 'ignore')
+        fname = re.sub('\W+', '_', fname.lower())
+        fname = '%s.%s.btpl' % (security.get_user().username, fname)
```

Comment:
Don't use usernames as uid : user names are unicode, can have special chars inside ('nng/apr'), are not unique and not persistent (can change during somebody's life)